integration-tests/e2e-tests/src/abilities/WalletSdk.ts

```diff
@@ -1,196 +1,157 @@
-    proofRequestStack: Message[]
+    credentialOfferStack: Message[];
+    issuedCredentialStack: Message[];
+    proofRequestStack: Message[];
   }) => Promise<void>): Interaction {
     return Interaction.where("#actor uses wallet sdk", async actor => {
       await callback(WalletSdk.as(actor).sdk, {
         credentialOfferStack: WalletSdk.as(actor).messages.credentialOfferStack,
         issuedCredentialStack: WalletSdk.as(actor).messages.issuedCredentialStack,
         proofRequestStack: WalletSdk.as(actor).messages.proofRequestStack
-      })
-    })
+      });
+    });
   }
 
   async discard(): Promise<void> {
-    await this.sdk.stop()
+    await this.sdk.stop();
   }
 
   async initialise(): Promise<void> {
     this.sdk.addListener(
       ListenerKey.MESSAGE, (messages: SDK.Domain.Message[]) => {
         for (const message of messages) {
-          this.messages.enqueue(message)
+          this.messages.enqueue(message);
         }
       }
-    )
+    );
 
-    await this.sdk.start()
+    await this.sdk.start();
   }
 
   isInitialised(): boolean {
-    return this.sdk.state != "stopped"
+    return this.sdk.state != "stopped";
   }
 }
 
 class WalletSdkBuilder {
   private static async getMediatorDidThroughOob(): Promise<string> {
-    const response = await axios.get(CloudAgentConfiguration.mediatorOobUrl)
-    const encodedData = response.data.split("?_oob=")[1]
-    const oobData = JSON.parse(Buffer.from(encodedData, "base64").toString())
-    return oobData.from
+    const response = await axios.get(CloudAgentConfiguration.mediatorOobUrl);
+    const encodedData = response.data.split("?_oob=")[1];
+    const oobData = JSON.parse(Buffer.from(encodedData, "base64").toString());
+    return oobData.from;
   }
 
   static async createInstance() {
-    const apollo = new Apollo()
-    const castor = new Castor(apollo);
-    const store = new InMemoryStore()
+    const apollo = new Apollo();
+    const store = new InMemoryStore();
     const pluto = new SDK.Pluto(store, apollo);
-    await pluto.start()
-
-    const api = new ApiImpl()
-    const didcomm = new DIDCommWrapper(apollo, castor, pluto)
-    const mercury = new Mercury(castor, didcomm, api)
-
-    const mediatorDID = Domain.DID.fromString(await WalletSdkBuilder.getMediatorDidThroughOob())
-    const mediatorStore = new PublicMediatorStore(pluto)
-
-    const mediatorHandler = new BasicMediatorHandler(
-      mediatorDID,
-      mercury,
-      mediatorStore,
-    )
-
-    const connectionsManager = new ConnectionsManager(
-      castor,
-      mercury,
-      pluto,
-      mediatorHandler,
-    )
-
-    const seed = apollo.createRandomSeed().seed
-    return new Agent(
-      apollo,
-      castor,
-      pluto,
-      mercury,
-      mediatorHandler,
-      connectionsManager,
-      seed,
-    )
+    const mediatorDID = Domain.DID.fromString(await WalletSdkBuilder.getMediatorDidThroughOob());
+
+    return Agent.initialize({ apollo, pluto, mediatorDID });
```

Comment:
this is the fix for e2e, allowing the Agent to handle it's creation which correctly connects the dependencies

Reply:
what do you mean? are you saying that using the full constructor won't work?

Reply:
great question, that's not why I've changed this, was just avoiding dependency hell.
but have just checked and it does appear to be the case.
I'd suggest we raise a separate bug ticket as it's not related to this PR